docs(changelog): add breaking changes for v0.41.0

adharsh277 opened the pull request:
<!--  Thanks for sending a pull request! Here are some tips for you:
1. If this is your first time, please read our contributor guidelines in the https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md file.
2. Please label this pull request according to what type of issue you are addressing.
3. Please add a release note!
4. If the PR is unfinished while opening it specify a wip in the title before the actual title, for example, "wip: my awesome feature"
-->

**What type of PR is this?**

/kind documentation

**Any specific area of the project related to this PR?**

/area docs

**What this PR does / why we need it**:

This PR updates the documentation to reflect breaking changes introduced in Falco 0.41.0. It summarizes deprecated CLI flags, container metadata plugin updates, and runtime behavior changes mentioned in issue #3497.

**Which issue(s) this PR fixes**:

Fixes #3497

**Special notes for your reviewer**:

- Based on the tracking info in issue #3497
- Added notes on plugin behavior and deprecated flags

**Does this PR introduce a user-facing change?**

```release-note
BREAKING CHANGE: Deprecated CLI flags (-S, -A, -b) have been removed. Container metadata is now handled via a plugin and affects musl builds. Output fields for containers and k8s namespaces have new names.

poiana commented:
Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

poiana commented:
@adharsh277: The label(s) `area/docs` cannot be applied, because the repository doesn't have them.

<details>

In response to [this](https://github.com/falcosecurity/falco/pull/3545):

>
>
>**What type of PR is this?**
>
>/kind documentation
>
>**Any specific area of the project related to this PR?**
>
>/area docs
>
>**What this PR does / why we need it**:
>
>This PR updates the documentation to reflect breaking changes introduced in Falco 0.41.0. It summarizes deprecated CLI flags, container metadata plugin updates, and runtime behavior changes mentioned in issue #3497.
>
>**Which issue(s) this PR fixes**:
>
>Fixes #3497
>
>**Special notes for your reviewer**:
>
>- Based on the tracking info in issue #3497
>- Added notes on plugin behavior and deprecated flags
>
>**Does this PR introduce a user-facing change?**
>
>```release-note
>BREAKING CHANGE: Deprecated CLI flags (-S, -A, -b) have been removed. Container metadata is now handled via a plugin and affects musl builds. Output fields for containers and k8s namespaces have new names.
> 
> 
>
>


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/3545#" title="Author self-approved">adharsh277</a>*
**Once this PR has been reviewed and has the lgtm label**, please assign [andreagit97](https://github.com/andreagit97) for approval. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/falco/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["andreagit97"]} -->

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [4675270](https://github.com/falcosecurity/falco/commits/4675270a4d3d0d2529b436121022fcddb15d616b) docs(changelog): add breaking changes for v0.41.0

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
Welcome @adharsh277! It looks like this is your first PR to falcosecurity/falco 🎉

adharsh277 commented:
I've updated the commit to include the necessary Signed-off-by: line and added the RELEASE NOTE block as you requested. Please let me know if you need anything else!

FedeDP commented:
Thanks for this PR! As shared here: https://github.com/falcosecurity/falco/issues/3497#issuecomment-2850074548 there is no need to manage breaking changes ourselves; the CHANGELOG file is auto-generated and will be updated right before the Falco release!
Anyway, thanks for opening this one and trying to help :)

FedeDP commented:
Closing :)
/close

poiana commented:
@FedeDP: Closed this PR.

<details>

In response to [this](https://github.com/falcosecurity/falco/pull/3545#issuecomment-2853514166):

>Closing :)
>/close


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>
